src/main.rs

```diff
@@ -120,29 +129,24 @@ fn startup_system(
 // Player 2 -> UP -> 'W' Key
 // Player 2 -> DOWN -> 'S' Key
 #[allow(clippy::type_complexity)]
-fn keyboard_input_system(
-    mut players: QuerySet<(
-        Query<&mut Position, With<LeftPlayer>>,
-        Query<&mut Position, With<RightPlayer>>,
-    )>,
-    key: Res<Input<KeyCode>>,
-) {
-    // TODO Rewrite using 'direction' and 'paddle.speed' instead of linear increments
-    for mut left_position in players.q0_mut().iter_mut() {
-        if key.pressed(KeyCode::W) {
-            left_position.y += 1.0;
-        } else if key.pressed(KeyCode::S) {
-            left_position.y -= 1.0;
-        }
-        left_position.y = left_position.y.min(50.0).max(-50.0);
-    }
-    for mut right_position in players.q1_mut().iter_mut() {
-        if key.pressed(KeyCode::Up) {
-            right_position.y += 1.0;
-        } else if key.pressed(KeyCode::Down) {
-            right_position.y -= 1.0;
+fn keyboard_input_system(mut players: Query<(&mut Position, &Player)>, key: Res<Input<KeyCode>>) {
```

Comment:
If you wanted to query for `LeftPlayer` or `RightPlayer` using bevy, it would be wrapped the component with `Option<T>`

https://bevy-cheatbook.github.io/programming/queries.html

```rust
fn check_zero_health(
    // access entities that have `Health` and `Transform` components
    // get read-only access to `Health` and mutable access to `Transform`
    // optional component: get access to `Player` if it exists
    mut query: Query<(&Health, &mut Transform, Option<&Player>)>,
) {
    // get all matching entities
    for (health, mut transform, player) in query.iter_mut() {
        eprintln!("Entity at {} has {} HP.", transform.translation, health.hp);

        // center if hp is zero
        if health.hp <= 0.0 {
            transform.translation = Vec3::ZERO;
        }

        if let Some(player) = player {
            // the current entity is the player!
            // do something special!
        }
    }
}

```